src/test.js

```diff
@@ -737,23 +741,41 @@ function internalStart( test ) {
 	}
 }
 
-function numberOfTests( module ) {
-	let count = module.tests.length;
+function collectTests( module, key ) {
+	let tests = [].concat( module[ key ] );
 	const modules = [ ...module.childModules ];
 
 	// Do a breadth-first traversal of the child modules
 	while ( modules.length ) {
 		const nextModule =  modules.shift();
-		count += nextModule.tests.length;
+		tests = tests.concat( nextModule[ key ] );
```

Comment:
`tests.push.apply( tests, nextModule[ keys ] );`

it avoids creating a new array and discarding the previous one.